src/trudp.c

```diff
@@ -182,6 +185,23 @@ void trudpChannelDestroyAddr(trudpData *td, const char *addr, int port, int chan
         trudpChannelData *tcd = (trudpChannelData *)teoMapGet(td->map, (const uint8_t*)key, key_length, NULL);
 
         if (tcd != NULL && tcd != (void *)-1) {
+            if (td->channel_key == tcd->channel_key) {
+                teoMapIterator it;
+                teoMapElementData *el;
+                uint32_t min_timeout_sq = UINT32_MAX;
+                uint64_t current_time = teoGetTimestampFull();
+                teoMapIteratorReset(&it, tcd->td->map);
+                while((el = teoMapIteratorNext(&it))) {
+                    trudpChannelData *tcd = (trudpChannelData *)teoMapIteratorElementData(el, NULL);
+                    uint32_t timeout_sq = trudpChannelSendQueueGetTimeout(tcd, current_time);
+                    if(timeout_sq < min_timeout_sq) {
+                    min_timeout_sq = timeout_sq;
+                    tcd->td->expected_max_time = timeout_sq;
+                    tcd->td->channel_key = tcd->channel_key;
+                    }
+                    if(!min_timeout_sq) break;
+                }
```

Comment:
Code like this exist in two places - here and in trudp_channel.c:635, would much better to extract it to function